### PR TITLE
Extract code useful for financial assistance loading into helpers.py …

### PR DESCRIPTION
**Pull request by catherinedevlin**
WORK IN PROGRESS 

From helpers.py, they can be used by contracts loader or the financial assistance loader.

Also some formatting changes (using isort and yapf) and refactoring of colon-splitting into a single function.

### CODE REVIEW

File: usaspending_api/awards/management/commands/helpers.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        for word in code_str.split():
+            query_set = query_set.filter(country_name__icontains=word)
+        country_code = query_set.first()
+
+    return country_code
+
+
+def get_or_create_award(row):
+    piid = row.get("piid", None)
+    parent_award_id = row.get("idvpiid", None)
```
Comment (afrasier) [inline usaspending_api/awards/management/commands/helpers.py]: This stuff is specific to Procurement/Contracts. May needs to add some things to grab fain/uri or a separate method for financial assistance

File: usaspending_api/awards/management/commands/helpers.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+from usaspending_api.references.models import Agency, Location, RefCountryCode
+
+
+def up2colon(input_string):
+    'Takes the part of a string before `:`, if any.'
+
+    return input_string.split(':')[0].strip()
+
+
+def get_agency(self, agency_string):
```
Comment (afrasier) [inline usaspending_api/awards/management/commands/helpers.py]: Whoops, missed this. 'self' needs to be removed here.

File: usaspending_api/awards/management/commands/helpers.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
+from datetime import datetime
+
+from django.db.models import Q
+
+from usaspending_api.awards.models import Award
+from usaspending_api.references.models import Agency, Location, RefCountryCode
+
+
+def up2colon(input_string):
+    'Takes the part of a string before `:`, if any.'
```
Comment (afrasier) [inline usaspending_api/awards/management/commands/helpers.py]: This throws an error if `input_string` is None, might want to check that and just return None if input_string is none

I'm not exactly sure how/when/why the input is ever None, I think it has something to do with csvreader, but while testing this did come up.

File: usaspending_api/awards/management/commands/helpers.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    return input_string.split(':')[0].strip()
+
+
+def get_agency(self, agency_string):
+    agency_code = up2colon(agency_string)
+    agency = Agency.objects.filter(
+        subtier_agency__subtier_code=agency_code).first()
+    if not agency:
+        self.logger.error("Missing agency: " + agency_string)
```
Comment (afrasier) [inline usaspending_api/awards/management/commands/helpers.py]: self won't work here, also logger is not defined. Might want to pass in the logger or grab it when the function is called

File: usaspending_api/awards/management/commands/helpers.py (hunk 1 above)
Comment (catherinedevlin) [inline usaspending_api/awards/management/commands/helpers.py]: That's right, I'd forgotten about that.  I'll move this back to loadcontracts after all.

File: usaspending_api/awards/management/commands/helpers.py (hunk 3 above)
Comment (catherinedevlin) [inline usaspending_api/awards/management/commands/helpers.py]: Good point.  Here's a fix.

**Comment by catherinedevlin** [comment]
I ended up working from a different branch and creating a pull request from that instead: https://github.com/fedspendingtransparency/usaspending-api/pull/106.  Thanks for these comments, they were incorporated over there.

**Pull request closed without merging**